.github/workflows/release.yaml

```
@@ -0,0 +1,52 @@
+name: Continuous integration
+
+on:
+  push:
+    branches:
+      - master
+      - release-**
+  pull_request:
+    branches:
+      - '**'
+  release:
+    types: [published]
+
+jobs:
+
+  release:
+    runs-on: ubuntu-20.04
+    env:
+      OSGEO_USERNAME: ${{ secrets.OSGEO_USERNAME }}
+      OSGEO_PASSWORD: ${{ secrets.OSGEO_PASSWORD }}
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      TX_TOKEN: ${{ secrets.TX_TOKEN }}
+      GITHUB_REF: ${{ github.ref }}
+    if: ${{ github.event_name == 'release' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Install dependencies
+        run: |
+          sudo apt update && sudo apt install qttools5-dev-tools qt5-default
+          sudo pip install qgis-plugin-ci==2.0.1
+      - name: Release
+        run: |
+          qgis-plugin-ci release ${GITHUB_REF##*/v} --transifex-token ${TX_TOKEN} --github-token ${GITHUB_TOKEN} --osgeo-username ${OSGEO_USERNAME} --osgeo-password ${OSGEO_PASSWORD}
```

```suggestion
          qgis-plugin-ci release ${GITHUB_REF##*/v} --transifex-token ${{ secrets.TX_TOKEN }} --github-token ${GITHUB_TOKEN} --osgeo-username ${OSGEO_USERNAME} --osgeo-password ${OSGEO_PASSWORD}
```

Better use them in place, this decreases the risk of leaking them (e.g. by enabling https://github.com/marketplace/actions/debugging-with-tmate without ssh key enabled)

---

Done, but then can't you also access that through bash command history ? would it maybe be better to still have then as env vars, but just on that particular step ?

---

If it's required to be an env var for a particular step, then it can be done as well. As general best practice I just try to reduce the scope to as little/precise as possible.

About bash command history, I don't think so (that normally doesn't remember scripts, only manually entered commands) but I didn't check.